src/main/java/uk/gov/hmcts/reform/notificationservice/service/NotificationService.java

```diff
@@ -39,14 +41,14 @@ public void processPendingNotifications() {
         notifications.forEach(this::processNotifications);
     }
 
-    private ErrorNotificationRequest mapToRequest(Notification notification) {
-        return new ErrorNotificationRequest(
-            notification.zipFileName,
-            notification.poBox,
-            notification.errorCode.name(),
-            notification.errorDescription,
-            String.valueOf(notification.id)
-        );
+    @Transactional(readOnly = true)
+    public List<NotificationResponse> findNotificationsByFileNameAndService(String fileName, String service) {
```

Comment:
perhaps `findByFileNameAndService` would be enough as it is `notificationService` after-all